tests/test_main.py

```diff
@@ -15,3 +17,38 @@ def runner() -> CliRunner:
 def test_main_fails(runner: CliRunner) -> None:
     result = runner.invoke(__main__.main)
     assert result.exit_code == 2
+
+
+def test_retrieve_called_with_auth_headers(
+    request: pytest.FixtureRequest, runner: CliRunner
+) -> None:
+    if request.config.getoption("typeguard_packages"):
+        pytest.skip(
+            "The import hook that typeguard uses seems to break the mock below."
+        )
+
+    expected_headers = {
+        "User-Agent": "python-requests/2.32.3",
```

Comment:
Is this value going to need to change if the installed version of `requests` changes?

Reply:
Yes — I don't love it either, thanks for flagging. I think we can probably consolidate to do a single version check (from your other comment) and build it that way.